### PR TITLE
Allow setting variables via `--set`

### PR DESCRIPTION
**Pull request by Milo123459**
This allows you to set variables to a service. For example:

```
railway variables --set "HELLO=WORLD" --set "MULTIPLE=AT_ONCE"
```

### CODE REVIEW

File: src/commands/variables.rs [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
+    environment: Option<String>,
+
     /// Show variables in KV format
     #[clap(short, long)]
     kv: bool,
+
+    /// The "{key}={value}" environment variable pair to set the service variables.
+    /// Example:
+    ///
+    /// ```bash
```
Comment (coffee-cup) [inline src/commands/variables.rs:33]: The quotes shouldn't be here as it doesn't render properly with `--help`

image

File: src/commands/variables.rs [hunk 2]
```diff
@@ -24,53 +43,88 @@ pub async fn command(args: Args, json: bool) -> Result<()> {
...
+            configs.get_backboard(),
+            vars,
         )
-    } else {
-        return Err(RailwayError::NoServiceLinked.into());
-    };
+        .await?;
+
+        spinner.finish_with_message(format!("Set {fmt_variables}"));
+        return Ok(());
```
Comment (coffee-cup) [inline src/commands/variables.rs:119]: Suggestion: Can this be extracted out to its own function. We could then also have another function to parse and format the variables from the args that is testable

Comment (Milo123459) [reply src/commands/variables.rs:119]: I don't think that would be useful. This logic only applies to the --set part.

Comment (ndneighbor) [reply src/commands/variables.rs:119]: Less so about the usefulness in the command writ large but more so, it's easier from a maintenance perspective to have the function separate so if you or the team adds this elsewhere, we don't have to chase down a bespoke implementation.

Comment (Milo123459) [reply src/commands/variables.rs]: Fixed

File: src/commands/variables.rs [hunk 3]
```diff
@@ -91,8 +105,64 @@ pub async fn command(args: Args, json: bool) -> Result<()> {
...
+    spinner.enable_steady_tick(Duration::from_millis(100));
+    let vars = mutations::variable_collection_upsert::Variables {
+        project_id: project,
+        environment_id,
+        service_id,
+        variables,
+    };
+    post_graphql::<mutations::VariableCollectionUpsert, _>(client, configs.get_backboard(), vars)
+        .await?;
+    spinner.finish_with_message(format!("Set {fmt_variables}"));
```
Comment (coffee-cup) [inline src/commands/variables.rs:166]: Very big nit
```suggestion
    spinner.finish_with_message(format!("Set variables {fmt_variables}"));
```

**Review by coffee-cup — APPROVED** [review]

**Pull request merged**